aptos-move/framework/aptos-framework/sources/dora_committee.move

```diff
@@ -0,0 +1,1042 @@
+/// Design:
+/// CommitteeInfoStore: store all the committee information, supported operation: add, remove
+/// CommitteeInfo: store all the dora node information, supported operation: add, remove
+/// DoraNodeInfo: store the dora node information, supported operation: update
+/// NodeData: store the dora node information with operator's address
+/// requirements:
+/// 1. two committee has different id
+/// 2. two indentical node should not be in the same committee
+module aptos_framework::dora_committee {
+    use std::error;
+    use std::option;
+    use std::signer;
+    use std::vector;
+    use aptos_std::capability;
+    use aptos_framework::event::{emit_event, EventHandle};
+    use aptos_std::simple_map::{Self, SimpleMap};
+    use aptos_framework::account::{Self, new_event_handle};
+
+    /// The number of committee is not equal to the number of committee member
+    const INVALID_COMMITTEE_NUMBERS: u64 = 4;
+
+    /// The node is not found in the committee
+    const NODE_NOT_FOUND: u64 = 5;
+
+    /// The committee is not found
+    const INVALID_COMMITTEE_ID: u64 = 6;
+
+    /// The committee type is invalid
+    const INVALID_COMMITTEE_TYPE: u64 = 7;
+
+    /// The number of nodes in the committee is invalid
+    const INVALID_NODE_NUMBERS: u64 = 8;
+
+    /// Define the CommitteeType as constants
+    const FAMILY: u8 = 1;
+    const CLAN: u8 = 2;
+    const TRIBE: u8 = 3;
+
+    const SEED_COMMITTEE: vector<u8> = b"aptos_framework::dora_committee::CommitteeInfoStore";
+
+    /// Capability that grants an owner the right to perform action.
+    struct OwnerCap has drop {}
+
+    struct SupraCommitteeEventHandler has key {
+        create: EventHandle<CreateCommitteeInfoStoreEvent>,
+        add_committee: EventHandle<AddCommitteeEvent>,
+        remove_committee: EventHandle<RemoveCommitteeEvent>,
+        update_committee: EventHandle<UpdateCommitteeEvent>,
+        add_committee_member: EventHandle<AddCommitteeMemberEvent>,
+        remove_committee_member: EventHandle<RemoveCommitteeMemberEvent>,
+        update_node_info: EventHandle<UpdateNodeInfoEvent>,
+        update_dkg_flag: EventHandle<UpdateDkgFlagEvent>,
+    }
+
+    struct CommitteeInfoStore has key {
+        // map from unique committee id -> CommitteeInfo
+        committee_map: SimpleMap<u64, CommitteeInfo>,
+        // map from node address to committee to which the node belongs
+        node_to_committee_map: SimpleMap<address, u64>,
+    }
+
+    struct CommitteeInfo has store, drop, copy {
+        // map from node address to DoraNodeInfo which provides other details pertaining to the node
+        map: SimpleMap<address, DoraNodeInfo>,
+        // a flag that indicates whether the committee already has performed DKG successfully or not, thus indicating whether corresponding public key are valid to be used for threshold signing
+        has_valid_dkg: bool,
+        committee_type: u8
```

Comment:
is there any view function to get `committee_type` from `committee_id`?
I think it should be add in `get_committee_info` view function
```
{
    node_data: vector<NodeData>,
    committee_type: u8
}
```

Reply:
Will do in #7. Thank you